sparse_strips/vello_cpu/src/dispatch/single_threaded.rs

```diff
@@ -61,13 +111,217 @@ impl SingleThreadedDispatcher {
+                            region.x,
+                            region.y,
+                            *layer_id,
+                            bg,
+                            layer_manager,
+                            encoded_paints,
+                        );
+
+                        assert_eq!(
+                            fine.blend_buf.len(),
+                            1,
+                            "blend buffer should contain exactly one layer after tile processing"
+                        );
+
+                        fine.pack(region);
+                    });
+                }
+            }
+        }
+    }
+
+    /// Processes all rendering commands for a single layer within a specific tile.
+    ///
+    /// This handles the complex logic of composing filtered layers by:
+    /// 1. Running normal rendering commands in sequence.
+    /// 2. When encountering a filtered layer reference, compositing its pre-rendered
+    ///    content from the layer manager.
+    /// 3. Skipping the filtered layer's internal commands (already rendered separately).
+    ///
+    /// # Arguments
+    /// * `fine` - The fine rasterizer instance.
+    /// * `x`, `y` - Wide tile coordinates.
+    /// * `layer_id` - The layer being processed.
+    /// * `clear_color` - Initial color for the tile.
+    /// * `layer_manager` - Storage for filtered layer buffers.
+    /// * `encoded_paints` - Paint definitions for the scene.
+    fn process_layer_tile<S: Simd, F: FineKernel<S>>(
+        &self,
+        fine: &mut Fine<S, F>,
+        x: u16,
+        y: u16,
+        layer_id: u32,
+        clear_color: PremulColor,
+        layer_manager: &mut LayerManager,
+        encoded_paints: &[EncodedPaint],
+    ) {
+        let wtile = self.wide.get(x, y);
+        fine.set_coords(x, y);
+        fine.clear(clear_color);
+
+        // Process all commands in this layer's render range.
+        if let Some(ranges) = wtile.layer_cmd_ranges.get(&layer_id) {
+            let mut cmd_idx = ranges.render_range.start;
+            while cmd_idx < ranges.render_range.end {
+                let cmd = &wtile.cmds[cmd_idx];
+                fine.run_cmd(cmd, &self.strip_storage.alphas, encoded_paints);
+
+                // Special handling for filtered layer composition.
+                if let Cmd::PushBuf(LayerKind::Filtered(filtered_layer_id)) = cmd {
+                    // Check if the next command is a clip for this filtered layer.
+                    let next_cmd = &wtile.cmds[cmd_idx + 1];
+                    if let Cmd::PushBuf(LayerKind::Clip(_)) = next_cmd {
+                        fine.run_cmd(next_cmd, &self.strip_storage.alphas, encoded_paints);
+                        cmd_idx += 1;
+                    }
+
+                    // Composite the filtered layer's content from the layer manager.
+                    if let Some(mut region) =
+                        layer_manager.layer_tile_region_mut(*filtered_layer_id, x, y)
```

Comment:
Do we ever expect this to be `None`?

Reply:
Currently, we expect `push_buf` to be called for all wide tiles. However, optimizing this is already on my list: now that we’ve introduced the wide tile bounding box, we know exactly where the buffers should be allocated. I haven’t done this optimization here for the filter, since the same change will be required for other operations as well. Updating it should be done carefully and is better handled in a separate review/PR.